### ISSUE TITLE
ERROR: Error: Error parsing config: Unexpected token ; in JSON at position <position-number>

### ISSUE THREAD

**Issue by GeekyPeas** [vishaltelangre/music-dl#31]
I tried running the example `music-dl http://gaana.com/album/aashiqui` and I got the following error:

```
ERROR: Error: Error parsing config: Unexpected token ; in JSON at position <position-number>
```
for each of the .mp3 file. I tried to debug this by injecting several `console.log()` into the code. I found out with:

```
search.findLink().then((link) => console.log("1",link))
```
that the `YouTubeSearch` returned the correct URL. However when I tried to inject:
```
new YouTubeDownloader(link, outputPath, outputFilename).download()
.then((mp3OutputPath) => {
      console.log("HERE",mp3OutputPath)
      const tagger = new Id3Tagger(mp3OutputPath, trackInfo);
      tagger.tag();
      resolve(mp3OutputPath);
      }).catch(reject);
```
in `youtube/client` the above `console.log()` never outputed. Instead i got the above `ERROR` message. So I suspect this narrows down where the error is caused.

I am not good with `java-script`. Could you please debug and fix this?

**Comment by vishaltelangre** [comment]
Thanks, @GeekyPeas for reporting this issue and trying to debug it. I really appreciate your help. 

I will try to reproduce this issue at my and will try to fix it.

Can you please share more information about your environment such as which OS do you use with its version, output of `music-dl -v`, `npm -v` and `node -v`? Thanks.

**Comment by GeekyPeas** [comment]
The following are details of my OS:
```
NAME="Ubuntu"
VERSION="19.10 (Eoan Ermine)"
```

The other details are:
```
music-dl v0.0.8
npm v 6.14.6
node v 10.15.2
``` 

EDIT: I recently updated `node` to `v 12.18.3`. I am getting the same error:

```
$ music-dl http://gaana.com/album/aashiqui
(get) Remote end closed socket abruptly., retrying https://gaana.com/album/aashiqui (2/3)
(node:16750) [DEP0066] DeprecationWarning: OutgoingMessage.prototype._headers is deprecated
Album/Playlist: Aashiqui (/home/geek/aashiqui)
→ (1/10) Dheere Dhheere Se Meri Zindagi Mein Aana
(get) Remote end closed socket abruptly., retrying https://gaana.com/album/aashiqui (3/3)
1 https://www.youtube.com/watch?v=pKsD2GiQhOo
ERROR: Error: Error parsing config: Unexpected token ; in JSON at position 179391
→ (2/10) Ek Sanam Chahiye Aashiqui Ke Liye
1 https://www.youtube.com/watch?v=fBylcT-TWZw
ERROR: Error: Error parsing config: Unexpected token ; in JSON at position 75916
→ (3/10) Tu Meri Zindagi Hai
1 https://www.youtube.com/watch?v=oEg_iXEWlt4
ERROR: Error: Error parsing config: Unexpected token ; in JSON at position 76036
→ (4/10) Nazar Ke Samne
1 https://www.youtube.com/watch?v=A4_KJ8t-dbo
ERROR: Error: Error parsing config: Unexpected token ; in JSON at position 158393
→ (5/10) Main Duniya Bhula Doonga
1 https://www.youtube.com/watch?v=otQmzlm-s7Q
ERROR: Error: Error parsing config: Unexpected token ; in JSON at position 185493
→ (6/10) Jaan-e-zigar Jaaneman
1 https://www.youtube.com/watch?v=HFXuHJLDcVo
ERROR: Error: Error parsing config: Unexpected token ; in JSON at position 64732
→ (7/10) Ab Tere Bin
1 https://www.youtube.com/watch?v=rTatsmUgmiA
ERROR: Error: Error parsing config: Unexpected token ; in JSON at position 83728
→ (8/10) Mera Dil Tere Liye
1 https://www.youtube.com/watch?v=TD8I45cJA54
ERROR: Error: Error parsing config: Unexpected token ; in JSON at position 161054
→ (9/10) Dil Ka Aalam
1 https://www.youtube.com/watch?v=BaAoZA0fup0
ERROR: Error: Error parsing config: Unexpected token ; in JSON at position 69287
→ (10/10) Jaan-e-zigar Jaaneman (part-1)
1 https://www.youtube.com/watch?v=zAHXA58cqHY
ERROR: Error: Error parsing config: Unexpected token ; in JSON at position 298034
```

As you can see the program correctly gets the URL from YouTube. (I had injected  `console.log("1", link)` into `youtube/client` for this. I could reproduce the errors by `npm uninstall -g music-dl` and then installing again just to be sure that this error was not caused by any modifications at my end. Then I inserted the `console.log()` to check if the URL are correctly getting fetched.)

Please let me know if you like me to insert `console.log()` in my copy of `music-dl` to help you debug.

**Comment by vishaltelangre** [comment]
@GeekyPeas Thanks for the all information.

I have fixed this issue in https://github.com/vishaltelangre/music-dl/pull/32. 

Please reinstall this package using `npm install -g music-dl` and confirm the version using `music-dl -v` which should be `v0.0.9`.

Also, you will no longer need the YouTube's API key! 😄 

Please let me know whether it works or not. Closing this issue for now.

**Comment by GeekyPeas** [comment]
Great! This works. However `music-dl -v` or `music-dl -h` don't work any more. I think.

**Comment by vishaltelangre** [comment]
@GeekyPeas You've got an eagle's eye! Thanks for sharing the details about the regression. I fixed it in the latest version `v0.0.10`.

After all, looks like this package needs some basic unit tests.
